modules/social_features/social_core/social_core.module

```diff
@@ -107,9 +107,7 @@ function social_core_entity_form_display_alter(EntityFormDisplayInterface $form_
       $fields
     );
 
-    $form_display
-      ->setThirdPartySetting('field_group', 'group_settings', $field_groups['group_settings'])
-      ->save();
+    $form_display->setThirdPartySetting('field_group', 'group_settings', $field_groups['group_settings']);
```

Comment:
@navneet0693 could you elaborate a bit on why this is necessary?
so whenever a next person comes here they are not trying to chain `->save();` 

Also wondering if we would need to setComponents on the form_display instead of third party settings, see example below:

```
  if ($context['form_mode'] === 'layout_builder') {
    $form_display->setComponent('moderation_state', [
      'type' => 'moderation_state_default',
      'weight' => -900,
      'settings' => [],
    ]);
  }
```

Although if it work's I'd be happy with that but right now I have too little information to understand this :)

Reply:
> could you elaborate a bit on why this is necessary?

From Drupal core's `html/core/lib/Drupal/Core/Entity/EntityInterface.php` class.

```
/**
   * Saves an entity permanently.
   *
   * When saving existing entities, the entity is assumed to be complete,
   * partial updates of entities are not supported.
   *
   * @return int
   *   Either SAVED_NEW or SAVED_UPDATED, depending on the operation performed.
   *
   * @throws \Drupal\Core\Entity\EntityStorageException
   *   In case of failures an exception is thrown.
   */
  public function save();
```

which says that it will save the entities permanently in the database. Calling `->save()`in an alter function can lead to the following:

1. Saving unwanted settings permanently
2. Saving unwanted translations in the database

If someone wants to do it, he/she should do it directly in a hook_update_N.

> Also wondering if we would need to setComponents on the form_display instead of third party settings

This is not required as the code is only intended to move the already existing "other fields" in the Settings section, only if they are not part of any field group.

Reply:
Sorry, I meant inline in code, so the next developer looking at that code also understands why :D

Reply:
(But thanks for the elaboration, that makes it clear :) ! )